Automattic/vip-go-mu-plugins
# Pre-set the `apikey` portion of empty `parsely` site options to `site_url` (if loaded)

## Description


This hooks into the [`option_parsely` filter](https://developer.wordpress.org/reference/hooks/option_option/) when the wp-parsely plugin is loaded by this repo.

If an `apikey` is specified in the option, the filtering function does not alter the value.
If a `_wpvip_init_option` value is set in the option, the filtering function does not alter the value (this is set when this function runs for the first time while an `apikey` is not set).

During this initialization, we reference `site_url()` for a smart default for the `apikey`.

If `site_url()` includes a path beyond a simple `/` (e.g. for a subdirectory install), path parts are prepended to the domain to emulate a "subdomain" as slashes are not permitted in Parse.ly api keys. This is a standard workaround per @hbbtstar & @arhine.

## Changelog Description


### Plugin Updated: VIP Parse.ly Integration -- Pre-set API Keys

We made some behind-the-scenes changes that will help customers try Parse.ly on their sites more easily.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Check out PR.
1. Delete the `parsely` site option
1. Browse to WP-Admin > Settings > Parsely
1. Verify an appropriate value is present in the `Parse.ly Site ID` Settings field.
